src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1046,7 +1046,7 @@ def show_container_cd_url(resource_group_name, name, slot=None):
 
 def view_in_browser(resource_group_name, name, slot=None, logs=False):
     site = _generic_site_operation(resource_group_name, name, 'get', slot)
-    url = site.default_host_name
+    url = site.enabled_host_names[0]  # picks the custom domain URL incase a domain is assigned
```

Comment:
Does the webapp service guarantee the first one will be the customized one, rather the default one?  Please confirm the service behavior and add a test for that. The other option is to look for a hostname which is not the default one and also not the scm one.

Reply:
@yugangw-msft yes the API add the items such that the first item is the domain URL to use.
I have tested this on apps, slots & function apps with custom domain & SSL.

Reply:
Okay, let us keep like this then. If it ever gets broken, you own me a test.

Reply:
Sure you got it! If this fails you can tell me 'I told you so'